backend/db/refresh.py

```diff
@@ -36,46 +36,49 @@ def refresh_db(
     time will not be fetched, resulting in an incomplete database. Therefore by default this will raise an error unless
     this is set to True."""
     local = use_local_database
-    print('INFO: Starting database refresh.', flush=True)  # flush: for gh action
     t0, t0_str = datetime.now(), current_datetime()
 
-    if check_db_status_var('refresh_status') == 'active':
-        update_db_status_var('new_request_while_refreshing', t0_str, local)
-        return
+    with get_db_connection(local=local) as con:
+        last_refresh = last_refresh_timestamp(con)
+        if since and dp.parse(since) > dp.parse(last_refresh) and not force_non_contiguity:
+            raise ValueError(SINCE_ERR)
+        since = since if since else last_refresh
 
-    update_db_status_var('refresh_status', 'active', local)
-    update_db_status_var('last_refresh_request', current_datetime(), local)
-    try:
-        with get_db_connection(local=local) as con:
-            last_refresh = last_refresh_timestamp(con)
-            if since and dp.parse(since) > dp.parse(last_refresh) and not force_non_contiguity:
-                raise ValueError(SINCE_ERR)
-            since = since if since else last_refresh
+        if not is_new_db_data(since):
+            'INFO: Database is up to date, no refresh necessary'
```

Comment:
I like this change! Kinda silly I didn't add it before.

I mentioned some ideas in [this other comment](https://github.com/jhu-bids/TermHub/pull/469#discussion_r1256447069).

I didn't label these options in my comment, but I'll call this option "A":
You could simply determine if there's any data during the normal call to `fetch_cset_and_member_objects()`.

```py
def csets_and_members_enclave_to_db(
    con: Connection, schema: str, since: Union[datetime, str], filter_0_member_sets=True
):
    """Fetch new csets and members, if needed, and then update database with them."""
    print('Fetching new data from the N3C data enclave...')
    t0 = datetime.now()
    csets_and_members: Dict[str, List[Dict]] = fetch_cset_and_member_objects(since)
```

After that last line, you could check if there's any new data, and if not, simply return `None`. This will return execution back to `refresh_db()` in `refresh.py`, where you could check if `None` is returned, and then exit out.

Option B:
I proposed an implementation of `is_new_db_data()`. You could use that and keep where you're currently calling `is_new_db_data()` in `refresh.py`.

Or Option C:
Maybe you'll think of something I haven't thought of!